### PR TITLE
[Android] Fix rounding / type issue when adding VSYNC offset

### PR DESCRIPTION
**Pull request by peak3d**
## Description
int64_t += ( (long long) / (float) ) lead to strange results when adding 2 VSYNC offsets.
int64_t += static_cast<int64_t >( (long long) / (float) ) solve the issue

## Motivation and Context
Stutter during video playback

## How Has This Been Tested?
Play "moving bars" samples in different fps on matching / multiple monitor Hz.
WETEK Hub / AFTV Gen 2

## Types of change
- [X] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

### CODE REVIEW

**Comment by peak3d** [comment]
build issues unrealated, correct @Rechi ?

**Comment by Rechi** [comment]
jenkins build this please

**Comment by da-anda** [comment]
will test the nightly this evening to see if my odd playback sync issues are also fixed by this PR

**Comment by iz8mbw** [comment]
@peak3d @Rechi @da-anda

Hi guys,
this PR solve stuttering during video playback (and many many thanks for that) but, may be, it introduce a problem when skip the video during the playback.
I have putted in play a mkv (1080p at 24fps) and skipped between chapters: move forward and move backward.
Often happens the video and audio freeze but the time of the movie goes on from where I skipped.
To simulate open a mkv that contains chapters and skip forward and backward between chapters.
Tested the most recent nightly build on Nvidia Shield TV with Andorid 8 and latest firmware.
Thanks.

**Comment by peak3d** [comment]
@iz8mbw can you pls. provide a debug log with "video" component logging enabled? Thx!

**Comment by iz8mbw** [comment]
@peak3d sure, give me some times. Thanks!

**Comment by iz8mbw** [comment]
@peak3d
Hi.
I'll provide log asap, sorry for delay.
What I can say to you is that only one mkv movie with DTS audio track gives me that problem. I should test better.
Others movie are OK. The seeking between chapters is more fast and reactive for movies with Dolby Digital track than DTS movies.
Anyway that yours is one of the best PR in the last months.
Since this PR finally I have a perfect audio/video sync, previously only movies with DTS audio track were in sync, now also movies with Dolby Digital tracks are in perfect sync.
Also the time needed to starts the playback is more fast.
Now no more video stuttering in 3:2 pulldown scenario. Previously to this PR to have no stuttering I was forced to sync refresh rate to match video, now with this PR I have smooth playback also in 3:2 pulldown scenario (24fps sent in a 60Hz signal)

In conclusion, since the good work and the very good results of thst PR, I can also have this new minor bug during seeking.

@peak3d @Rechi @da-anda many thanks for your work in Kodi!

**Comment by iz8mbw** [comment]
@peak3d @Rechi @da-anda

Hi.
Here the Kodi log: https://paste.ee/p/tj0R4

What I did:

1. Started Kodi with Debug log active (also with Video component as requested)
2. Put in play a mkv (1080p with DTS 5.1 channels audio track)
3. Skip (seek) forward to Chapter 2: Audio OK, Video freezed for some seconds to latest displayed frame, after some seconds video started again
4. Skip (seek) forward to Chapter 3: Audio MUTE, Video always FREEZED and never recovered
5. Stop playback via GUI
6. Closed Kodi

So, the problem on point 3 of my list is annoing but after some seconds it's OK.
The problem on point 4 of my list is "the big" problem.

Many thanks for your support.

**Comment by Rechi** [comment]
@iz8mbw why are you pinging me all the time, I have nothing to do with this PR!

**Comment by iz8mbw** [comment]
@Rechi very sorry.

**Comment by iz8mbw** [comment]
@peak3d any news about that issue (dropping packet)?

**Comment by peak3d** [comment]
@iz8mbw @fritsch did some changes which depends on audio / dt issues, are you updating your device regulary?

**Comment by iz8mbw** [comment]
What do you mean?

**Comment by peak3d** [comment]
Updating kodi

**Comment by iz8mbw** [comment]
Using 20181017 version on Nvidia Shield, problem persist

**Comment by peak3d** [comment]
ok, will investigate

**Comment by iz8mbw** [comment]
Thanks @peak3d issue solved.
